### PR TITLE
Removing raw component pointer caching

### PR DESCRIPTION
**Pull request by raktalen**
Removing raw component pointer caching that was causing a dangling pointer access

## What does this PR do?
Fixes https://github.com/o3de/o3de/issues/14691

## How was this PR tested?
Tested in the editor

### CODE REVIEW

**Review by forkercat — APPROVED** [review]

**Comment by michabr** [comment]
I'm curious when/why the component becomes invalid. Is it possible to also clear it out at the right time?

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DPEComponentAdapter.h [hunk 1]
```diff
@@ -58,8 +58,7 @@ namespace AZ::DocumentPropertyEditor
     protected:
         AZStd::string m_componentAlias;
         AZ::EntityId m_entityId;
-
-        AZ::Component* m_componentInstance = nullptr;
+        AZ::ComponentId m_componentId;
```
[the remark this answers is not in the document]
Comment (raktalen) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DPEComponentAdapter.h:61]: Good call

**Comment by raktalen** [comment]
> 



> I'm curious when/why the component becomes invalid. Is it possible to also clear it out at the right time?

I did look into that approach as well, but honestly we just shouldn't be caching raw pointers if we don't need them.

**Comment by michabr** [comment]
> > 
> 
> > I'm curious when/why the component becomes invalid. Is it possible to also clear it out at the right time?
> 
> I did look into that approach as well, but honestly we just shouldn't be caching raw pointers if we don't need them.

I agree with not caching raw pointers. Just wondering when/why they become invalid without being reset.

**Comment by lumberyard-employee-dm** [comment]
> > > 
> > 
> > 
> > > I'm curious when/why the component becomes invalid. Is it possible to also clear it out at the right time?
> > 
> > 
> > I did look into that approach as well, but honestly we just shouldn't be caching raw pointers if we don't need them.
> 
> I agree with not caching raw pointers. Just wondering when/why they become invalid without being reset.

It is fine to Cache Components, but only between the Activate and Deactivate the entity that the Component is attached to.

If anything the DPEComponentAdapter could cache components in response to the [EntityBus](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Component/EntityBus.h#L160-L174) `OnEntityActivated` and `OnEntityDeactivated`

**Comment by rgba16f** [comment]
Seems like this change got merge bombed, it's hard to see the actual change from the 1000 modified files it shows now.

**Pull request closed without merging**